android_mobile/src/main/java/com/alexstyl/specialdates/donate/DonateActivity.java

```diff
@@ -81,13 +83,13 @@ public void onClick(View v) {
             @Override
             public void run() {
                 scrollToDonate();
-
             }
 
             private void scrollToDonate() {
                 CoordinatorLayout.LayoutParams params = (CoordinatorLayout.LayoutParams) appBarLayout.getLayoutParams();
                 AppBarLayout.Behavior behavior = (AppBarLayout.Behavior) params.getBehavior();
-                behavior.onNestedFling(coordinator, appBarLayout, null, 0, 50, true);
+                if (behavior != null)
```

Comment:
Hey, Actually the NPE error is totally misleading. There is nothing wrong with how behavior is initialized. The real cause of this issue, is that the in-app billing is not supported in emulator. So `onDonateException` is called finishing the activity before `behavior` gets initialized thus resulting to NPE on **some** versions of android (e.g api 19). 

So I'd suggest that you merge this pull request as is for now and later on decide how you wanna handle this rare case.

Check out the attached images. First one shows the NPE crash from null `behavior` that occurs without the if-null check. and second image shows the in-app-billing error which however 
doesn't result to a crash after the if-null check has been applied to `behavior`

![npe](https://cloud.githubusercontent.com/assets/4888330/25878863/96da7056-3536-11e7-86b9-3e280c801fe4.png)

![iaperror](https://cloud.githubusercontent.com/assets/4888330/25878871/a33b2f7a-3536-11e7-9fe4-d1881446aa8d.png)